src/expressions/operator-expr.md

```diff
@@ -478,15 +478,48 @@ unsafe {
 assert_eq!(values[1], 3);
 ```
 
-#### Slice DST pointer to pointer cast
+#### Sized to sized pointer casts
 
-For slice types like `[T]` and `[U]`, the raw pointer types `*const [T]`, `*mut [T]`,
-`*const [U]`, and `*mut [U]` encode the number of elements in this slice. Casts between
-these raw pointer types preserve the number of elements. Note that, as a consequence,
-such casts do *not* necessarily preserve the size of the pointer's referent (e.g.,
-casting `*const [u16]` to `*const [u8]` will result in a raw pointer which refers to an
-object of half the size of the original). The same holds for `str` and any compound type
-whose unsized tail is a slice type, such as struct `Foo(i32, [u8])` or `(u64, Foo)`.
+A sized pointer to pointer cast `*const T as *const U` or `*mut T as *mut U`
+where `T: Sized` and `U: Sized` produces a value identical to the original
+pointer.
+
+#### Unsized to unsized pointer casts
+
+Any unsized to unsized pointer cast `*const T as *const U` or `*mut T as *mut U`
+produces a value identical to the original pointer if `T` and `U` are compatible
+unsized types.
```

Comment:
This then raises the question, what does it do if they are not compatible?

Reply:
Do we have to answer that now? I'd like to avoid pulling as many threads as possible. 😅

Reply:
The PR in its current form introduces a confusing new sentence. So yes this needs to be fixed.

But I think the structure I propose will also handle this pretty naturally.

Reply:
Is the sentence "compatible unsized types"? If so, that's not my phrasing. It's borrowed from the table just above.

Reply:
Ah, I see. But in the table it is clear that if the types are not compatible, then the cast is simply not permitted.

Reply:
Ok, so I've removed the double reference to validity since it's already covered by the table and rewrote it to only focus on behavior. Please check it out!